pkg/solana/transmitter.go

```diff
@@ -81,18 +80,18 @@ func (c *ContractTracker) Transmit(
 
 	// Send transaction, and wait for confirmation:
 	go func() {
-		txSig, err := confirm.SendAndConfirmTransactionWithOpts(
+		txSig, err := c.client.rpc.SendTransactionWithOpts(
 			context.Background(), // does not use libocr transmit context
-			c.client.rpc,
-			c.client.ws,
 			tx,
-			true, // skip preflight
+			false, // use preflight as first check
 			rpc.CommitmentConfirmed,
 		)
 
 		if err != nil {
 			c.lggr.Errorf("error on Transmit.SendAndConfirmTransaction: %s", err.Error())
 		}
+		// TODO: poll rpc for tx confirmation (WS connection unreliable)
+		// TODO: see if SendTransactionWithOpts returns quickly enough to be used outside of goroutine
```

Comment:
We should not count on it being quick enough, as this is a side-effect that depends on other systems and infrastructure (connection).